### ISSUE TITLE
command line interface

### ISSUE THREAD

**Issue by RNAer** [biocore/micronota#1]
Need suggestions on use cases to determine the interface. Please, any feedback is welcome. This is what I suggest:

* main command
```
micronota
--help
--version
--debug
--verbose
--info Print settings and external exe version
```
* subcommand for database management
```
micronota db download
micronota db create
micronota db validate
micronota db list
```
* subcommand for format conversion. Conversion between fasta, GenBank, GFF3
```
micronota convert
--in Input file path
--out Output file path
```

* subcommand for annotation
```
micronota annotate
--transl_table  Genetic code
--cpus          Number of CPUs to use
--min_len       Minimum seq length to be annotated
--evalue        Similarity e-value cut-off (default '1e-06')
--kingdom       Bacteria or Archaea
--force         Force overwriting
--input         Either directory (all fasta files inside) or fasta files
--output_dir    Output directory
--output_fmt    Output format. GenBank, GFF3, etc.
--config        Parse config file for the run
--meta          Switch on/off metagenome mode
```
config file is used to fine tune the parameters for external exe as advanced users like. This config file will be parsed with configparser module. Example:
```
[prodigal]
mode = normal
trans_table = 11

[cmsearch]
cpu = 12
incE = 0.01  # e-value threshold
```

Things to think about:
1. would you like to annotate a specific feature (like only ncRNA or prophage) only? Do we want to enable to run each external software individually?

2. would you like to take a annotated GenBank file and re-annotate or add some other types of annotation? i.e. incrementally annotate a genome.

3. what other types of tasks you think is very useful?

**Comment by ElDeveloper** [comment]
Should the config file for micronota be at the top level of the command? Or are these only really useful when annotating?

**Comment by RNAer** [comment]
yes, I only imagined the use cases for external tools to wrap. not sure for
the top level so far. I don't think putting it at the top level changes
anything, though. We can do that.

**Comment by mortonjt** [comment]
Not sure about the `--turn-on` and `--turn-off` parameters ...
